tests/csapi/apidoc_room_receipts_test.go

```diff
@@ -45,7 +49,35 @@ func TestRoomReceipts(t *testing.T) {
 	alice.MustDo(t, "POST", []string{"_matrix", "client", "v3", "rooms", roomID, "receipt", "m.read", eventID}, client.WithJSONBody(t, struct{}{}))
 
 	// Make sure the read receipt shows up in sync.
-	alice.MustSyncUntil(t, client.SyncReq{}, syncHasReadReceipt(roomID, alice.UserID, eventID))
+	sinceToken := alice.MustSyncUntil(t, client.SyncReq{}, syncHasReadReceipt(roomID, alice.UserID, eventID))
+
+	// Receipt events include a `room_id` field over federation, but they should
+	// not do so down `/sync` to clients. Ensure homeservers strip that field out.
+	t.Run("Receipts DO NOT include a `room_id` field", func(t *testing.T) {
+		// Send another event to read.
+		eventID2 := sendMessageIntoRoom(t, alice, roomID)
+
+		// Send a read receipt for the event.
+		alice.MustDo(t, "POST", []string{"_matrix", "client", "v3", "rooms", roomID, "receipt", "m.read", eventID2}, client.WithJSONBody(t, struct{}{}))
+
+		alice.MustSyncUntil(
+			t,
+			client.SyncReq{Since: sinceToken},
+			client.SyncEphemeralHas(roomID, func(r gjson.Result) bool {
+				if r.Get("type").Str != "m.read" {
+					return false
+				}
+
+				// Ensure that the `room_id` field does NOT exist.
+				if r.Get("room_id").Exists() {
+					t.Fatalf("Read receipt included `room_id` field down sync: %s", r.Raw)
```

Comment:
```suggestion
					t.Fatalf("Read receipt should not contain 'room_id' field when syncing but saw: %s", r.Raw)
```

I think this better explains that this is unexpected/erroneous, etc.

(also applies below)